src/coin_test/data/dataset_generator.py

```diff
@@ -0,0 +1,153 @@
+"""Define the SyntheticDatasets class and subclasses."""
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pandas as pd
+
+from .datasets import CustomDataset
+
+
+class DatasetGenerator(ABC):
+    """Create synthetic datasets."""
+
+    @abstractmethod
+    def generate(
+        self, timedelta: pd.Timedelta, seed: int | None = None, n: int = 1
```

Comment:
For time lengths, we should probably use `pd.Timedelta | pd.DateOffset`. Mostly interchangeable, except that you can't compare lengths against each other. Hopefully an easy change if possible.

Reply:
Done